### DISCUSSION TITLE
how to setup custom oauth 2.0 password grant in next auth ?

### DISCUSSION THREAD

**Discussion by 360synertech-dev1** [nextauthjs/next-auth#3702] [Help]
### Question 💬

I want to know the best way to implement the password grant in next auth. So I can use session.
I try to use custom Oauth but its only support the { grant_type: "authorization_code" },

I am using Laravel password as a backend api and next js as a frontend app

### How to reproduce ☕️

export default NextAuth({
  providers: [
    {
      id: "custom_oauth",
      name: "custom-oauth",
      type: "oauth",
      version: "2.0",
      scope: "*",
      state: true,
      protection: "state",
      params: { grant_type: "password" },
      idToken: true,
      authorizationUrl:
        "http://127.0.0.1:8000/oauth/token",
      accessTokenUrl: "http://127.0.0.1:8000/oauth/token",
      requestTokenUrl: "http://127.0.0.1:8000/oauth/token",
      clientId: process.env.CLIENT_ID,
      clientSecret: process.env.CLIENT_SECRET,
    },
  ],

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
There are no plans supporting it as it is now considered a legacy solution

https://oauth.net/2/grant-types/password/

**Reply by azadeh-syt** [reply]
Hi,
How about resource owner password credentials grant type? Can nextAuth support it?

**Reply by rf-exp** [reply]
Did you find a solution ?

**Reply by azadeh-syt** [reply]
Nope! Had to refactor my server.
